ssz_serialization/codec.nim

```diff
@@ -225,9 +225,6 @@ macro initSszUnionImpl(RecordType: type, input: openArray[byte]): untyped =
 
           enumInstanceSerializedFields(caseObj, fieldName, field):
             when fieldName != `SelectorFieldNameLit`:
-              if `input`.len <= 1:
-                raiseMalformedSszError(`type recordDef`, "invalid empty for selector")
-
               readSszValue(`input`.toOpenArray(1, `input`.len - 1), field)
```

Comment:
This does look quite strange to me as `input`.len can be 1, but apparently in that case this does return as an empty `openArray`, which is the goal.

Not sure if that's a fair practice, if not, I can change to make it more explicit.

Reply:
Considering we do often use `toOpenArray`(0, len -1)` type of construction, this is practically the same but from index 1, so I'll leave it in.